src/core/modules/GridLinesMgr.cpp

```diff
@@ -1129,8 +1195,60 @@ void SkyLine::draw(StelCore *core) const
 			if (alt<= 2*M_PI_180) i_min =static_cast<int>(-alt*M_180_PI)+2;
 			if (alt>=-2*M_PI_180) i_max-=static_cast<int>( alt*M_180_PI)+2;
 		}
-
-		for (int i=0; i<i_max; ++i)
+		if (line_type==ECLIPTIC_WITH_DATE)
+		{
+			// we must adapt (rotate) some labels to observers on the southern hemisphere.
+			const bool southernHemi = core->getCurrentLocation().getLatitude() < 0.f;
+			const float extraTextAngle = southernHemi ? M_PI_2f : -M_PI_2f;
+			const float shifty = (southernHemi ? -1.f : 0.25) *  static_cast<float>(sPainter.getFontMetrics().height());
+			const double currentFoV=core->getMovementMgr()->getCurrentFov();
+
+			// This special line type does not show the actual ecliptic line but only the partitions. These must be read from the precomputed static array eclipticOnDatePartitions
+			QMap<Vec3d, QString>::const_iterator it=eclipticOnDatePartitions.constBegin();
+			while (it != eclipticOnDatePartitions.constEnd())
+			{
+				double lng=it.key()[0]; // ecl. longitude, radians
+				const double nutation=it.key()[1]; // nutation in longitude, radians
+				const double aberration=it.key()[2]; // aberration, radians
+				if (core->getUseNutation())
+					lng+=nutation;
+				if (core->getUseAberration())
+					lng+=aberration;
+				const QString &label=it.value();
+				// draw and labels: derive the irregular tick lengths from labeling
+				Vec3d start=fpt;
+				Vec3d end= (label.length()>0) ? part10 : part1;
+				if (label.contains("5"))
+					end=part5;
+				Vec3d end10=part10;
+
+				const Mat4d& rotDay = Mat4d::rotation(partZAxis, lng);
+				start.transfo4d(rotDay);
+				end.transfo4d(rotDay);
+				end10.transfo4d(rotDay);
+
+				sPainter.drawGreatCircleArc(start, end, Q_NULLPTR, Q_NULLPTR, Q_NULLPTR);
+
+				if ((label.length()>0) && (
```

Comment:
There's way too many parentheses here. At least the single logical expressions should be simpler, i.e. instead of `(x)` or `((x))` it should be just `x`. Like so:
```cpp
if (label.length()>0 && (
					currentFoV<60. // all labels
					|| (currentFoV<=180. && !label.contains("5"))) // 1.MM/10.MM/20.MM
					|| label.startsWith("1.") // in any case
					)
```
Also, the indentation is misleading: the final opening parenthesis on the first line is closed on the third line after `contains("5")`, which is not obvious at a glance from the indentation structure.

Moreover, the compiler (g++-9) hints that the `&&` and the last `||` are on the same level, which is even less obvious at a glance, where the two `||` seem to be on the same level. Maybe the whole complicated expression should be split into several `bool` variables that would then be more simply combined in the `if` condition.

Reply:
No, this was indeed a logic error (without visible consequences). Too much of reformulating the rules... Thanks.